### PR TITLE
Deployment integration 

### PR DESCRIPTION
**Pull request by skofman1**
1. Deployment to Octopus on Staging.Runner 
2. IIS integration
3. Entity framework integration (to run migrations)

@xavierdecoster @maartenba

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,165 +1,166 @@
...
+
+# Build results
+
+[Dd]ebug/
+[Rr]elease/
+x64/
+#build/ # We use this folder for our Build Scripts
+[Bb]in/
+[Oo]bj/
+*.dg
```
Comment (skofman1) [inline .gitignore]: Added *.dg to .gitignore

File: NuGet.config [hunk 2]
```diff
@@ -1,12 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
    <packageSources>
-    <add key="aspnetrelease" value="https://www.myget.org/F/aspnetrelease/api/v3/index.json" />
     <add key="myget" value="https://www.myget.org/F/nugetbuild/api/v3/index.json" />
```
Comment (skofman1) [inline NuGet.config]: Removed all private aspnet feeds (rc2 packages were published to nuget.org)

File: src/NuGet.Services.Staging.BackgroundWorkers/MessageListener/TopicMessageListener.cs [hunk 3]
```diff
@@ -40,7 +41,10 @@ public TopicMessageListener(IOptions<TopicMessageListenerOptions> options, ILogg
             _options = options.Value;
             _logger = logger;
 
-            var factory = MessagingFactory.CreateFromConnectionString(_options.ServiceBusConnectionString);
+            var factory = MessagingFactory.Create(
+                _options.ServiceBusUri,
```
Comment (skofman1) [inline src/NuGet.Services.Staging.BackgroundWorkers/MessageListener/TopicMessageListener.cs]: Changed the way service bus topic subscription is accessed. The previous code forced me to use a SAS token of the entire service bus, and this way the SAS token is per topic (safer).

File: src/NuGet.Services.Staging.BackgroundWorkers/project.json [hunk 4]
```diff
@@ -7,13 +7,13 @@
         "NuGet.Services.Staging.PackageService": "1.0.0-*",
         "Microsoft.Extensions.Options": "1.0.0-rc2-*",
         "Newtonsoft.Json": "8.0.3",
-        "NuGet.Resolver": "3.5.0-*",
+        "NuGet.Resolver": "3.5.0-beta-final",
```
Comment (skofman1) [inline src/NuGet.Services.Staging.BackgroundWorkers/project.json]: using * to get the newest NuGet.* packages breakes me twice a day.. and also:  https://github.com/aspnet/EntityFramework/issues/5411

File: src/NuGet.Services.Staging.Manager/project.json [hunk 5]
```diff
@@ -43,9 +54,14 @@
     "tools": {
         "Microsoft.EntityFrameworkCore.Tools": {
             "imports": [ "portable-net451+win8" ],
-            "version": "1.0.0-rc2-*"
+            "version": "1.0.0-preview1-*"
+        },
+        "Microsoft.AspNetCore.Server.IISIntegration.Tools": {
```
Comment (skofman1) [inline src/NuGet.Services.Staging.Manager/project.json]: needed to get IIS to work

File: src/NuGet.Services.Staging.Manager/project.json [hunk 6]
```diff
@@ -1,7 +1,20 @@
...
+            "include": [ "Config", "wwwroot", "web.config", "appsettings.json" ]
+        }
+    },
+    "publishOptions": {
+        "include": [ "Config", "wwroot", "appsettings.json", "web.config" ]
+    },
+
+    "runtimeOptions": {
+        "gcServer": true,
+        "gcConcurrent": true
```
Comment (maartenba) [inline src/NuGet.Services.Staging.Manager/project.json]: Curious, any reason why these have to be added explicitly?

**Comment by maartenba** [comment]
LGTM, but can you maybe squash the commits a bit?

**Comment by skofman1** [comment]
Squashed and merged

**Pull request closed without merging**